addons/account/models/account_invoice.py

```diff
@@ -1120,13 +1120,25 @@ def _track_subtype(self, init_values):
     def _get_tax_amount_by_group(self):
         self.ensure_one()
         res = {}
+        result = {}
+        vals = {}
         currency = self.currency_id or self.company_id.currency_id
         for line in self.tax_line_ids:
             res.setdefault(line.tax_id.tax_group_id, 0.0)
             res[line.tax_id.tax_group_id] += line.amount
-        res = sorted(res.items(), key=lambda l: l[0].sequence)
-        res = map(lambda l: (l[0].name, l[1]), res)
-        return res
+        for inv_line in self.invoice_line_ids:
+            for tax_group in inv_line.invoice_line_tax_ids.mapped('tax_group_id'):
```

Comment:
No need to iterate over **invoice_line_ids** to find the **base** amount on which tax is applied. You can use **base** amount from **tax_line_ids** .